pkg/sparrow/targets/remote/gitlab/gitlab.go

```diff
@@ -242,98 +248,166 @@ func (c *client) PutFile(ctx context.Context, body remote.File) error { //nolint
-		}
-	}(resp.Body)
+	defer func() {
+		err = errors.Join(err, resp.Body.Close())
+	}()
 
+	// Unfortunately the Gitlab API does not give any information about what went wrong and just returns a 400 Bad Request in case of
+	// an error while committing the file. A probable cause for this is that the branch does not exist or is protected.
+	// The documentation documents some more possible errors, but not all of them: https://docs.gitlab.com/ee/api/repository_files.html#update-existing-file-in-repository
+	// Therefore we just check for the status code and return an error if it is not 200 OK.
+	// This is not ideal, but the best we can do with the current API without implementing a full blown error handling mechanism.
 	if resp.StatusCode != http.StatusCreated {
-		log.Error("Failed to post file", "status", resp.Status)
+		log.ErrorContext(ctx, "Failed to post file", "status", resp.Status)
 		return fmt.Errorf("request failed, status is %s", resp.Status)
 	}
 
 	return nil
 }
 
-// DeleteFile deletes the file matching the filename from the configured
-// gitlab repository
+// DeleteFile deletes the file matching the filename from the configured gitlab repository
 func (c *client) DeleteFile(ctx context.Context, file remote.File) error { //nolint:gocritic // no performance concerns yet
 	log := logger.FromContext(ctx).With("file", file)
 
 	if file.Name == "" {
 		return fmt.Errorf("filename is empty")
 	}
 
-	log.Debug("Deleting file from gitlab")
+	log.DebugContext(ctx, "Deleting file from gitlab")
 	n := url.PathEscape(file.Name)
-	b, err := file.Bytes()
+	b, err := file.Serialize(c.config.Branch)
 	if err != nil {
-		log.Error("Failed to create request", "error", err)
+		log.ErrorContext(ctx, "Failed to create request", "error", err)
 		return err
 	}
 
 	req, err := http.NewRequestWithContext(ctx,
 		http.MethodDelete,
-		fmt.Sprintf("%s/api/v4/projects/%d/repository/files/%s", c.baseUrl, c.projectID, n),
+		fmt.Sprintf("%s/api/v4/projects/%d/repository/files/%s", c.config.BaseURL, c.config.ProjectID, n),
 		bytes.NewBuffer(b),
 	)
 	if err != nil {
-		log.Error("Failed to create request", "error", err)
+		log.ErrorContext(ctx, "Failed to create request", "error", err)
 		return err
 	}
 
-	req.Header.Add("PRIVATE-TOKEN", c.token)
+	req.Header.Add("PRIVATE-TOKEN", c.config.Token)
 	req.Header.Add("Content-Type", "application/json")
 
-	resp, err := c.client.Do(req) //nolint:bodyclose // closed in defer
+	resp, err := c.client.Do(req)
 	if err != nil {
-		log.Error("Failed to delete file", "error", err)
+		log.ErrorContext(ctx, "Failed to delete file", "error", err)
 		return err
 	}
 
-	defer func(Body io.ReadCloser) {
-		err = Body.Close()
-		if err != nil {
-			log.Error("Failed to close response body", "error", err)
+	defer func() {
+		cErr := resp.Body.Close()
+		if cErr != nil {
+			log.ErrorContext(ctx, "Failed to close response body", "error", cErr)
+			err = errors.Join(err, cErr)
 		}
-	}(resp.Body)
+	}()
```

Comment:
Could we go for:

```go
	defer func() {
		err = errors.Join(err, resp.Body.Close())
	}()
```

as you did in the other code snippets? So that this case is always handled the same way?